packages/react-components/src/components/MessageThread.tsx

```diff
@@ -849,8 +849,12 @@ export const MessageThread = (props: MessageThreadProps): JSX.Element => {
       if (onLoadPreviousChatMessages) {
         isLoadingChatMessagesRef.current = true;
         // Fetch message until scrollTop reach the threshold for fetching new message
+        previousTopRef.current = chatScrollDivRef.current.scrollTop;
```

Comment:
Additional lines in this PR (852-857) looks a bit duplicate with line 878-892, do we have any concern of having same logic in 2 places?
I guess we can do this:
1. test if the logic could be in separated useEffect closures, if yes, then put them in different use effect (use effect will run after each loop of while, because rendering will be re-triggered), fetch function is responsible for only getting new messages without any dom operation
2. If we can't separate them into different useEffect, I guess we can remove 874-893, cause all the logic has been covered by your new code (your code condition is like a superset)

Reply:
You've found the core problem of the bug, just want to make the code a bit readable and without duplicate logic, please lemme know if you have anything unsure about my comment, would like to have a quick sync offline

Reply:
Thanks so much Nan!